app/controllers/reservations_controller.rb

```diff
@@ -295,10 +304,9 @@ def ability_resource
 
   # TODO you shouldn't be able to edit reservations that have passed or are outside of the cancellation period (check to make sure order has been placed)
   def invalid_for_update?
-    params[:id].to_i != @reservation.id ||
-    @reservation.actual_start_at ||
-    @reservation.actual_end_at ||
-    !(@reservation.can_edit? || @reservation.can_customer_edit?)
+    different_id = params[:id].to_i != @reservation.id
+    cannot_edit = current_user.administrator? ? !@reservation.can_edit? : !@reservation.can_customer_edit?
```

Comment:
I'm not sure this check needs to change. Also, we want to be careful with how we talk about "administrators". `User#administer?` is asking if the user is a global admin. Usually, we would want to use some other check like `manager_of` or something in the cancan abilities.